osutil/disks/disks_linux.go

```diff
@@ -632,6 +632,73 @@ func (d *disk) populatePartitions() error {
 			}
 			part.KernelDeviceNode = devname
 
+			// we should always have the partition type
+			partType, ok := udevProps["ID_PART_ENTRY_TYPE"]
+			if !ok {
+				return fmt.Errorf("cannot get udev properties for device %s (a partition of %s), missing required udev property \"ID_PART_ENTRY_TYPE\"", partDev, d.Dev())
+			}
+
+			// on dos disks, the type is formatted like "0xc", when we prefer to
+			// always use "0x0C" so fix the formatting
+			if d.schema == "DOS" {
+				partType = strings.TrimPrefix(partType, "0x")
+				v, err := strconv.ParseUint(partType, 16, 8)
+				if err != nil {
+					return fmt.Errorf("cannot get udev properties for device %s (a partition of %s), cannot convert MBR partition type %q: %v", partDev, d.Dev(), partType, err)
+				}
+				partType = fmt.Sprintf("%02X", v)
```

Comment:
```suggestion
				partType = fmt.Sprintf("%02x", v)
```
so that MBR and GPT type uuids are consistently lowercase

Reply:
I'm confused by the suggestion, this makes the partition type for DOS to be lower case, when we already make GPT UUID's upper case, so is your proposal that we should always use lower case or that we should use lower case for DOS and upper case for GPT? I think we should be consistent and I think we have precedent for other places using upper case already

Reply:
for example in the gadget.yaml for pi's we have:

```yaml
device-tree: bcm2709-rpi-3-b-plus
volumes:
  pi:
    schema: mbr
    bootloader: u-boot
    structure:
      - type: 0C
```

Reply:
In any case we were missing unit tests here and we were not consistently making the schema lower case, so I adjusted that and added some tests.